### ISSUE TITLE
Report error on component names not imported or provided

### ISSUE THREAD

**Issue by karlhorky** [mdx-js/mdx-analyzer#466]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

It would be helpful if component names which are not imported or provided to the current `.mdx` file are reported as errors.

Currently `Diagramxx` (typo) has the type `any`, and does not report any errors or warnings in VS Code:

```mdx
import Diagram from './diagram.mdx';

<Diagramxx />
```

![Screenshot 2024-08-12 at 09 40 32](https://github.com/user-attachments/assets/8d46da06-1999-4201-8815-97b75d556fe1)

This can create a confusing experience in MDX environments where this is not reported as a visible error, such as [ReMDX](https://github.com/nkzw-tech/remdx) (which uses Vite and [`compile()` from `@mdx-js/mdx`](https://github.com/nkzw-tech/remdx/blob/3cde2e303733632a1d58a512d45b0dca84234830/packages/vite-plugin-remdx/index.ts#L15-L28) - no error is visible in the browser or in the Vite CLI:

![Screenshot 2024-08-12 at 09 36 51](https://github.com/user-attachments/assets/8339d73c-5bff-4b4d-a146-cea050bf208b)

```bash
$ vite dev -- rest-apis

  VITE v5.4.0  ready in 526 ms

  ➜  Local:   http://localhost:5173/slide-decks/rest-apis/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
```

Only upon opening the DevTools is there an error visible:

![Screenshot 2024-08-12 at 09 53 27](https://github.com/user-attachments/assets/5120ebe8-78dc-4f08-ad43-8083b57c05e4)

Arguably, this could be better communicated in ReMDX and similar MDX environments, but still this is a runtime error, and it would improve the feedback loop to have an error in the editor.

---

I thought that maybe [the type of `global.MDXProvidedComponents`](https://github.com/mdx-js/mdx-analyzer#:~:text=Another%20alternative%20is%20to%20define%20the%20MDXProvidedComponents%20type%20globally.) would be consulted by MDX Analyzer, but setting this type does not appear to have an effect:

```ts
import type { MDXComponents } from 'mdx/types.js';

const components = {
  Other: () => JSX.Element;
} satisfies MDXComponents;

declare global {
  type MDXProvidedComponents = typeof components;
}
```

### Solution

Zero config reporting of problems (errors or warnings) on component names which have not been:

1. Imported in the local `.mdx` file
2. Provided via the `components` prop or global bundler provider configuration

Alternative: some configuration to allow for recognition of components

### Alternatives

Default configuration for `eslint-mdx` to report problems:

- https://github.com/mdx-js/eslint-mdx/issues/549

**Comment by karlhorky** [comment]
Creating a reproduction...

**Comment by karlhorky** [comment]
Oh, it must have been something in my monorepo project which caused the `global. MDXProvidedComponents` type to be not recognized 🙈

In the reproduction repo both `global. MDXProvidedComponents` and local `MDXProvidedComponents` type definition in `.mdx` files are working:

![Screenshot 2024-08-12 at 10 37 27](https://github.com/user-attachments/assets/4b7249a1-9d66-4aa4-82d6-a71530db1a16)

Components recognition feature looks good Remco, thanks! 🙌 Closing.
